src/pixelmapFeature.js

```diff
@@ -0,0 +1,451 @@
+ *   transformations for those two triangles.
+ * @returns {geo.pixelmapFeature}
+ */
+//////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////
+var pixelmapFeature = function (arg) {
+  'use strict';
+  if (!(this instanceof pixelmapFeature)) {
+    return new pixelmapFeature(arg);
+  }
+  arg = arg || {};
+  feature.call(this, arg);
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * @private
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  var m_this = this,
+      m_quadFeature,
+      m_srcImage,
+      m_info,
+      s_update = this._update,
+      s_init = this._init,
+      s_exit = this._exit;
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get/Set position accessor
+   *
+   * @returns {geo.pixelmap}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.position = function (val) {
+    if (val === undefined) {
+      return m_this.style('position');
+    } else if (val !== m_this.style('position')) {
+      m_this.style('position', val);
+      m_this.dataTime().modified();
+      m_this.modified();
+    }
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get/Set url accessor
+   *
+   * @returns {geo.pixelmap}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.url = function (val) {
+    if (val === undefined) {
+      return m_this.style('url');
+    } else if (val !== m_this.style('url')) {
+      m_srcImage = m_info = undefined;
+      m_this.style('url', val);
+      m_this.dataTime().modified();
+      m_this.modified();
+    }
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get the maximum index value from the pixelmap.  This is a value present in
+   * the pixelmap.
+   *
+   * @returns {geo.pixelmap}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.maxIndex = function () {
+    if (m_info) {
+      /* This isn't just m_info.mappedColors.length - 1, since there
+       * may be more data than actual indices. */
+      if (m_info.maxIndex === undefined) {
+        m_info.maxIndex = 0;
+        for (var idx in m_info.mappedColors) {
+          if (m_info.mappedColors.hasOwnProperty(idx)) {
+            m_info.maxIndex = Math.max(m_info.maxIndex, idx);
+          }
+        }
+      }
+      return m_info.maxIndex;
+    }
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get/Set mapColor accessor
+   *
+   * @returns {geo.pixelmap}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.mapColor = function (val) {
```

Comment:
In many features, we have certain styles also accessible by functions (position often is, even if we store it in style as well).  This allows for an easy side effect, such as only marking this as changed if it is actually different.

The problem with this approach in general is that we have different behaviors if these properties are set via style versus the special function -- for instance, `style('position', <value>)` won't update the dataTime and will always updated the modified time, even if the passed value is the same as the previous one.  We might want to refactor the feature style method so that it is easier to have convenience functions and the behavior is identical regardless of whether the convenience function or style is used.  However, that should be a general change, and not in this PR.